syntax = "proto3";

package graph;

service ServeGraph {
  rpc LoadCustomPnl (PNLPath) returns (NullArgument) {}
  rpc LoadScript (ScriptPath) returns (ScriptCompositions) {}
  rpc GetCompositions (NullArgument) returns (ScriptCompositions) {}
  rpc GetJSON (GraphName) returns (GraphJSON) {}
  rpc HealthCheck (NullArgument) returns (HealthStatus) {}
}

message NullArgument {

}

message HealthStatus {
  string status = 1;
}

message PNLPath {
  string path = 1;
}

message ScriptPath {
  string path = 1;
}

message ScriptCompositions {
   repeated string compositions = 1;
}

message GraphName {
  string name = 1;
}

message GraphJSON {
  string JSON = 1;
}
